pkg/cli/cvd.go

```diff
@@ -308,15 +318,65 @@ func (c *cvdCreator) createCVDFromLocalSrcs() ([]*hoapi.CVD, error) {
 	return res.CVDs, nil
 }
 
-func credentialsFactoryFromSource(source string) (CredentialsFactory, error) {
-	switch source {
+func credentialsFactoryFromSource(opts BuildAPICredentialsOpts) (CredentialsFactory, error) {
+	switch opts.Source {
 	case NoneCredentialsSource:
 		return func() string { return "" }, nil
 	case InjectedCredentialsSource:
 		return func() string { return client.InjectedCredentials }, nil
+	case FileCredentialSource:
+		return auth(opts.FilePath)
 	default:
-		return nil, fmt.Errorf("unknown credentials source: %s", source)
+		return nil, fmt.Errorf("unknown credentials source: %s", opts.Source)
+	}
+}
+
+type OAuthConfig interface {
+	auth() (*oauth2.Token, error)
+}
+
+type JwtConfig struct{ *jwt.Config }
+
+func NewJwtConfig(credential []byte) (*JwtConfig, error) {
+	jwtConfig, err := google.JWTConfigFromJSON(credential, "https://www.googleapis.com/auth/androidbuild.internal")
+	if err != nil {
+		return nil, err
+	}
+	return &JwtConfig{jwtConfig}, nil
+}
+
+func (c JwtConfig) auth() (*oauth2.Token, error) {
+	tk, err := c.TokenSource(context.Background()).Token()
+	if err != nil {
+		return nil, err
+	}
+	return tk, nil
+}
+
+// TODO: OAuth Config for end user
+
+func parseAuthConfig(credential []byte) (OAuthConfig, error) {
+	jwtConfig, err := NewJwtConfig(credential)
+	if err == nil {
+		return jwtConfig, nil
+	}
+	return nil, err
+}
```

Comment:
Do we need this function? IIUC, this function seems for obtaining `OAuthConfig` type object from `JwtConfig` type object. Couldn't we make it simpler without introducing a new function?

Reply:
The reason why I use a new function is because I'd like to implement 3LO auth flow for end-users use cases probably in next PR, which requires a different type of config. 

To answer the three questions together, the final code would look like this:
```
type OAuthConfig struct {*oauth2.Config}

func (c OAuthConfig) auth(*oauth2.Token, error) {
    // do 3LO auth flow
}

func parseAuthConfig(credential []byte) (OAuthConfig, error) {
	jwtConfig, err := NewJwtConfig(credential)
	if err == nil {
		return jwtConfig, nil
	}
        // if not jwtConfig, try to parse oauth2 config and return
        // else return parse error
}
```
So that the `func auth()` can do the authentication without considering which flow to use.

Does that make sense?

Reply:
I think it makes sense, but I prefer to change `OAuthConfig` interface type into the different name then. I think your purpose is having a interface type for common uses, and 2 struct types(`OAuthConfig` and `JwtConfig`) implements `auth()` for satisfying interface type. What about `AuthConfig` instead?

BTW, why don't we make a fallback mechanism on `func auth(filepath string)`? For instance, executing `JWTConfigFromJSON` and `TokenSource` in a single function called by `func auth(filepath string)`?